src/core/transpile.jl

```diff
@@ -649,13 +649,13 @@ function simplify_rz_gate(
     phase_angle::Real; 
     atol=1e-6)::Union{AbstractGate,Nothing}
     
-    if isapprox(phase_angle,π/2,atol=atol) 
+    if isapprox(phase_angle,π/2,atol=atol) || isapprox(phase_angle,-3*π/2,atol=atol)
```

Comment:
A more long term solution is to first determine the floating point modulo and then compare against that. This works for -3*pi/2, but fails for -7*pi/2.